tests/components/elkm1/test_config_flow.py

```diff
@@ -1741,3 +1766,232 @@ async def test_multiple_instances_with_tls_v12(hass: HomeAssistant) -> None:
         "username": "test-username",
     }
     assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_reconfigure_nonsecure(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test reconfigure flow switching to non-secure protocol."""
+    # Add mock_config_entry to hass before updating
+    mock_config_entry.add_to_hass(hass)
+
+    # Update mock_config_entry.data using async_update_entry
+    hass.config_entries.async_update_entry(
+        mock_config_entry,
+        data={
+            "auto_configure": True,
+            "host": "elk://localhost",
+            "username": "",
+            "password": "",
+            "prefix": "",
+        },
+    )
+
+    await hass.async_block_till_done()
+
+    result = await mock_config_entry.start_reconfigure_flow(hass)
+    await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "reconfigure"
+
+    # Mock the validation function and prevent reload
+    with (
+        patch(
+            "homeassistant.components.elkm1.config_flow.validate_input",
+            return_value={
+                "title": "Test Title",
+                CONF_HOST: "elk://1.2.3.4",
+                CONF_PREFIX: "",
+            },
+        ),
+        patch(
+            "homeassistant.components.elkm1.async_setup_entry",
+            return_value=True,
+        ) as mock_setup_entry,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "protocol": "non-secure",
+                "address": "1.2.3.4",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] is FlowResultType.ABORT
+    assert result2["reason"] == "reconfigure_successful"
+
+    # Verify the config entry was updated with the new data
+    assert dict(mock_config_entry.data) == {
+        "auto_configure": True,
+        "host": "elk://1.2.3.4",
+        "username": "",
+        "password": "",
+        "prefix": "",
+    }
+
+    # Verify the setup was called during reload
+    mock_setup_entry.assert_called_once()
+
+
+@pytest.mark.usefixtures("socket_enabled")
+async def test_reconfigure_tls(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test reconfigure flow switching to TLS 1.2 protocol."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.async_block_till_done()
+
+    result = await mock_config_entry.start_reconfigure_flow(hass)
+    await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "reconfigure"
+
+    # Mock the validation function and prevent reload
+    with (
+        patch(
+            "homeassistant.components.elkm1.config_flow.validate_input",
```

Comment:
Please don't patch the integration, instead patch the library code like the other tests. https://developers.home-assistant.io/docs/development_testing?_highlight=testing&_highlight=code#writing-tests-for-integrations